src/core/meshcop/dataset_manager.cpp

```diff
@@ -160,7 +160,7 @@ otError DatasetManager::Save(const Dataset &aDataset)
 
     if (isMasterkeyUpdated || compare > 0)
     {
-        mLocal.Save(aDataset);
+        SuccessOrExit(error = mLocal.Save(aDataset));
```

Comment:
logic changed here.

Reply:
Thanks @simonlingoogle, that's very careful of you!